src/Nethermind/Nethermind.Optimism/SimulateOptimismTransactionProcessor.cs

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Evm;
+using Nethermind.Evm.Tracing;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Logging;
+using Nethermind.State;
+using Nethermind.Facade.Simulate;
+
+namespace Nethermind.Optimism;
+
+public sealed class SimulateOptimismTransactionProcessor(
```

Comment:
I see that in the future this might require more implementation details. At the same time we create a processor type only to act on one flag: `validate` that could be embedded in the original `OptimismTransactionProcessor` as a special case (with the same flag). Have you considered this option?

Reply:
Before considering adding an additional flag to the original `OptimismTransactionProcessor` I thought of adding a decorator `SimulateTransactionProcessor(ITransactionProcessor inner)` but this cannot be easily done since we need to `override` methods that are not part of the interface (ex. `ShouldValidate`), and there is already a hierarchy implemented.

I think this change is the less invasive of all (it only removes the `sealed` modifier to `OptimismTransactionProcessor`) and it's consistent with the existing `SimulateTransactionProcessor`. Personally, I dislike modifying the original code to add special cases or using implementation inheritance to override behavior (unless there is a very good reason to do so).

Reply:
I think it's opinions. If you considered the flag, and don't like this approach, let's go with what is here.  Yes, the derivation structure in the simulation is a bit prohibitive to easily decorate it a bit more.